src/main/java/io/reactivex/rxjava3/internal/operators/single/SingleTimeout.java

```diff
@@ -113,11 +113,7 @@ public void onError(Throwable e) {
 
         @Override
         public void run() {
-            Disposable d = get();
-            if (d != DisposableHelper.DISPOSED && compareAndSet(d, DisposableHelper.DISPOSED)) {
```

Comment:
Are there more patterns in the codebase like this which would also need to be fixed?

Reply:
That's the only one with this pattern I could find (i.e., no loop).

Reply:
This is fine because it loops:
https://github.com/ReactiveX/RxJava/blob/5bb119b1489afe4fbc5bb51b828be95971ee8f1d/src/main/java/io/reactivex/rxjava3/internal/observers/FutureObserver.java#L50-L56

This is fine too:
https://github.com/ReactiveX/RxJava/blob/5bb119b1489afe4fbc5bb51b828be95971ee8f1d/src/main/java/io/reactivex/rxjava3/internal/observers/FutureMultiObserver.java#L49-L55

Reply:
:shipit: